### ISSUE TITLE
Currently testing golang rate limiter v9 have some failures 

### ISSUE THREAD

**Issue by luchoman08** [alicebob/miniredis#176]
When the rate limiter is tested with miniredis, that start to give some wrong values, a test implementation:

The redis implementation used in tests can be found here [https://github.com/go-redis/redis_rate/tree/v9](https://github.com/go-redis/redis_rate/tree/v9
)

```go
package cache

import (
	"context"
	"testing"

	"github.com/alicebob/miniredis/v2"

	"github.com/go-redis/redis/v8"
	"github.com/go-redis/redis_rate/v9"
)

func TestRateLimit300(t *testing.T) {
	mock, err := miniredis.Run()
	if err != nil {
		t.Error(err)
	}

	client := redis.NewClient(
		&redis.Options{
			Addr: mock.Addr(),
		})

	limit := redis_rate.NewLimiter(client)

	times := 300
	key := "limit-300"
	for i := 1; i < times + 1; i++ {
		res, err := limit.Allow(context.Background(), key, redis_rate.PerSecond(times))
		if err != nil {
			t.Error(err)
		}

		if times - i != res.Remaining {
			t.Errorf("Remaining wanted: %d, got: %d", times-i, res.Remaining)
		}
	}
}
```

The output seems like:

```
    limit_test.go:34: Remaining wanted: 296, got: 297
    limit_test.go:34: Remaining wanted: 295, got: 296
    limit_test.go:34: Remaining wanted: 294, got: 296
    limit_test.go:34: Remaining wanted: 293, got: 295
    limit_test.go:34: Remaining wanted: 292, got: 294
    limit_test.go:34: Remaining wanted: 291, got: 293
    limit_test.go:34: Remaining wanted: 290, got: 292
     .
     .
     .
    limit_test.go:34: Remaining wanted: 1, got: 43
    limit_test.go:34: Remaining wanted: 0, got: 42
```

As shown in the output, remaining value from miniredis starts to be greater that should be

`go version go1.15.2 darwin/amd64`

**Comment by alicebob** [comment]
On Sun, Oct 11, 2020 at 05:41:55PM -0700, luchoman08 wrote:
> When the rate limiter is tested with miniredis, that start to give some wrong values, a test implementation:

Interesting, I'll have a look soon.
Can you start with importing the "v2" version of miniredis?
"github.com/alicebob/miniredis/v2"

Thanks!

**Comment by alicebob** [comment]
> Closed #176.

That actually solved the problem? Good to know :)
